src/providers/hana/qgshanaconnection.cpp

```diff
@@ -0,0 +1,621 @@
+                          "IS_VALID = 'TRUE' AND SCHEMA_NAME IN (SELECT SCHEMA_NAME FROM SYS.SCHEMAS WHERE "
+                          "HAS_PRIVILEGES = 'TRUE') AND SCHEMA_NAME NOT LIKE_REGEXPR 'SYS|_SYS.*|UIS|SAP_XS|SAP_REST|HANA_XS') "
+                          "ON VIEW_OID = VIEW_OID_2 AND SCHEMA_NAME LIKE '%2' AND DATA_TYPE_NAME LIKE_REGEXPR '%3'" )
+          .arg(
+            userTablesOnly ? "CURRENT_USER" : "%",
+            schema.isEmpty() ? "%" : schema,
+            allowGeometrylessTables ? "" : "ST_GEOMETRY|ST_POINT" );
+    ResultSetRef rsViews = stmt->executeQuery( reinterpret_cast<const char16_t *>( sql.unicode() ) );
+    while ( rsViews->next() )
+    {
+      QgsHanaLayerProperty layerProperty;
+      layerProperty.schemaName = QgsHanaUtils::toQString( rsViews->getNString( 1 ) );
+      layerProperty.tableName = QgsHanaUtils::toQString( rsViews->getNString( 2 ) );
+      layerProperty.geometryColName = ( QgsHanaUtils::toQString( rsViews->getNString( 4 ) ) != "ST_GEOMETRY" ) ? ""
+                                      : QgsHanaUtils::toQString( rsViews->getNString( 3 ) );
+      layerProperty.tableComment = QgsHanaUtils::toQString( rsViews->getNString( 6 ) );
+      layerProperty.isView = true;
+
+      addNewLayer( list, layerProperty, layerProperty.geometryColName.isEmpty() );
+    }
+    rsViews->close();
+
+    int size = list.size();
+    for ( int i = 0; i < size; ++i )
+    {
+      QgsHanaLayerProperty &lp1 = list[i];
+      bool found = false;
+      for ( int j = 0; j < size; ++j )
+      {
+        if ( i != j )
+        {
+          const QgsHanaLayerProperty &lp2 = list.at( j );
+          if ( lp1.schemaName == lp2.schemaName && lp1.tableName == lp2.tableName )
+          {
+            found = true;
+            break;
+          }
+        }
+      }
+
+      if ( !found )
+        lp1.isUnique = true;
+    }
+  }
+  catch ( const Exception &ex )
+  {
+    throw QgsHanaException( ex.what() );
+  }
+  return list;
+}
+
+void QgsHanaConnection::readLayerInfo( QgsHanaLayerProperty &layerProperty )
+{
+  layerProperty.srid = getLayerSRID( layerProperty );
+  layerProperty.type = getLayerGeometryType( layerProperty );
+  layerProperty.pkCols = getLayerPrimaryeKeys( layerProperty );
+}
+
+QVector<QgsHanaSchemaProperty> QgsHanaConnection::getSchemas( const QString &ownerName )
+{
+  QVector<QgsHanaSchemaProperty> list;
+  QString sql = QStringLiteral( "SELECT SCHEMA_NAME, SCHEMA_OWNER FROM SYS.SCHEMAS WHERE "
+                                "HAS_PRIVILEGES = 'TRUE' AND SCHEMA_OWNER LIKE '%1' AND "
+                                "SCHEMA_NAME NOT LIKE_REGEXPR 'SYS|_SYS.*|UIS|SAP_XS|SAP_REST|HANA_XS|XSSQLCC_' AND "
+                                "SCHEMA_NAME LIKE '%2'" )
+                .arg( ownerName.isEmpty() ? QStringLiteral( "%" ) : ownerName, mUri.schema().isEmpty() ? QStringLiteral( "%" ) : mUri.schema() );
```

Comment:
Use prepared statements. Here the escaping of `ownerName` and the `mUri.schema()` is missing. In prepared statements, escaping is not necessary.

Reply:
Done in 84beb39.